evm/src/cpu/kernel/asm/bignum/addmul.asm

```diff
@@ -0,0 +1,99 @@
+// Arithmetic on little-endian integers represented with 128-bit limbs.
+// All integers must be under a given length bound, and are padded with leading zeroes.
+
+// Sets a[0:len] += b[0:len] * val, and returns the carry.
```

Comment:
Hm, can't a carry be a "digit" (limb) too, not just a bit? In grade-school math the carry is a base-10 digit

Reply:
I overwhelmingly see "carry" to mean "carry bit", especially in low-level and assembly programming, but it's true that is a narrow perspective. The important thing here is to document what the range of the return value is, so feel free to call it simply "carry" but it'd be good to have a few extra words to explain that it is the size of a bignum limb, to avoid any ambiguity.

Reply:
I changed it to "carry limb" everywhere, I think that's the best of both worlds? So it's clear that it's functioning as the carry but there's no confusion that it's just one bit